include/ginkgo/core/distributed/matrix.hpp

```diff
@@ -536,6 +586,20 @@ class Matrix
                     ptr_param<const LinOp> local_matrix_template,
                     ptr_param<const LinOp> non_local_matrix_template);
 
+    // build local only matrix from existed data
+    explicit Matrix(std::shared_ptr<const Executor> exec,
+                    mpi::communicator comm, dim<2> size,
+                    std::shared_ptr<LinOp> local_linop);
```

Comment:
What is this used for? If you are using only a local linop without communication (i.e. no send/recv sizes, offsets, etc.) then you can just use the linop directly.